src/track/macroaction.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+namespace proto = mixxx::track::io;
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available, but that
+/// is subject to change.
+class MacroAction {
+  public:
+    enum class Type : uint32_t {
+        Jump = 0
+    };
+
+    MacroAction(double sourceFramePos, double targetFramePos)
+            : sourceFrame(sourceFramePos), targetFrame(targetFramePos), type(Type::Jump){};
+
+    Type getType() const {
+        return type;
+    }
+
+    double getSourcePosition() const {
+        return sourceFrame;
+    }
+    double getTargetPosition() const {
+        return targetFrame;
+    }
+
+    double getSourcePositionSample() const {
```

Comment:
maybe it makes sense to use the helper functions from `mixxx::audio::SignalInfo` instead of adding these methods?

Reply:
Yeah I intended to remove it, but it would introduce a lot of repetitious calls doing exactly what this method does, so in the end I decided to add it back in.